app/tasks_routes.py

```diff
@@ -0,0 +1,172 @@
+import datetime
+from app import db
+from app.models.task import Task
+from flask import Blueprint,jsonify,abort,make_response,request
+import requests
+from datetime import datetime
+import os
+from dotenv import load_dotenv
+
+tasks_bp = Blueprint('tasks_bp', __name__, url_prefix='/tasks')
+TASK_ID_PREFIX = '/<task_id>'
+
+def validate_task(task_id):
+    try:
+        task_id = int(task_id)
+    except:
+        abort(make_response({"message":f"Task {task_id} invalid"}, 400))
+
+    task = Task.query.get(task_id)
+
+    if not task:
+        abort(make_response({"message":f"Task {task_id} not found"}, 404))
+    
+    return task
+
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+
+    if "title" not in request_body or "description" not in request_body:
+        return make_response({"details": "Invalid data"}, 400)
+
+    new_task = Task(
+        title = request_body["title"],
+        description = request_body["description"],
+    )
+
+    db.session.add(new_task)
+    db.session.commit()
+    
+    return make_response({"task":new_task.to_dict()}, 201)
+
+@tasks_bp.route(TASK_ID_PREFIX,methods=['PUT'])
+def update_task(task_id):
+    request_body = request.get_json()
+    if "title" not in request_body or "description" not in request_body:
+        return make_response("Invalid Request, Title & Description Can't Be Empty", 400)
+
+    task = validate_task(task_id)
+    task = Task.query.get(task_id)
+    
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+
+    db.session.commit()
+    return make_response({"task":task.to_dict()}, 200)
+
+
+@tasks_bp.route(TASK_ID_PREFIX, methods=["GET"])
+# GET /task/id
+def handle_task(task_id):
+    # Query our db to grab the task that has the id we want:
+    task = validate_task(task_id)
+    task = Task.query.get(task_id)
+
+    if task.goal_id is not None:
+        return{"task": task.to_dict_in_goal()}
+    else:
+        return {"task": task.to_dict()}
```

Comment:
👍  See comment in `tasks` class that combines `to_dict_in_goal` and `to_dict`. 

with that refactor we can reduce these lines to:

```python
return {"task": task.to_dict()}
```